### ISSUE TITLE
bug: False positive on resetting all state when in a Hook

### ISSUE THREAD

**Issue by KyleBastien** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#42]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.5.5

### ESLint version

8.57.0

### Describe the bug

I have a relatively simple hook which debounces a value via a useEffect. The lint rule says that I'm hitting the resetting all state case, which I think is true, but also I don't think there is anything I can do for this case, because it's not in a Component, it's in a hook.

### Steps To Reproduce

1. Create a new file (e.g., useDebounce.ts)
2. Write a hook which might look like this:

```ts
/**
 * A custom hook that debounces a value with a specified delay
 * @param value - The value to debounce
 * @param delay - The delay in milliseconds
 * @returns The debounced value
 */
export function useDebounce<T>(value: T, delay: number): T {
  const [debouncedValue, setDebouncedValue] = useState<T>(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
}
```

3. See lint error: Avoid resetting all state when a prop changes. If "value" is a key, pass it as `key` instead so React will reset the component.

### Expected Behavior

I think in this case, since this is a custom hook, I don't think there is a way to use `key` here to avoid this effect. And I think the effect is necessary because of the cleanup action that is needed.

**Comment by KyleBastien** [comment]
Awesome! Thank you!
